jib-core/src/test/java/com/google/cloud/tools/jib/http/FailoverHttpClientTest.java

```diff
@@ -25,15 +28,21 @@
 import com.google.api.client.http.HttpTransport;
 import com.google.cloud.tools.jib.api.LogEvent;
 import com.google.cloud.tools.jib.blob.Blobs;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
```

Comment:
I do know this will never be an issue in practice at all, but we don't really want to introduce `com.sum.*` following general guidelines.

https://www.oracle.com/java/technologies/faq-sun-packages.html
https://rules.sonarsource.com/java/RSPEC-1191

Maybe take a look at `WithServerFailoverHttpClientTest.java` for an alternative.

Reply:
@chanseokoh this is not an issue, http server is exported and "public" even if the package does not make it that explicit: https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/master/src/jdk.httpserver/share/classes/module-info.java. I saw WithServerFailoverHttpClientTest but thought that since HttpServer is there and fits better the test it was more relevant than a socket layer solution which is less readable IMHO.

Reply:
I am aware of that and the package is exported in the module but still wanted to exclude it. Can you guarantee that, especially for Java 8, every JDK implementation (e.g., IBM JVM) provides this package? Even the Javadoc you linked still says "JDK-specific HTTP server API."

However, practically this will cause no issue to anyone, so I don't want to make your life too difficult. I can accept this.